services/feeCalculator/feeCalculator.go

```diff
@@ -270,3 +294,59 @@ func calcLitecoinFee(inputCount, outputCount, feePerByte int) int {
 func calcBitcoinCashFee(inputCount, outputCount, feePerByte int) int {
 	return (inputCount*148 + outputCount*34 + 10) * feePerByte
 }
+
+func speedControl(t interface{}, speed string) (*feeCalculator, int) {
+	var f feeCalculator
+	var gaz int
+
+	switch t.(type) {
+	case responses.BitcoinFeeResponse:
+		typed := t.(responses.BitcoinFeeResponse)
+		f.CalcFee = calcBitcoinFee
+		switch speed {
+		case "1":
+			f.FeePerByte = typed.HourFee
+		case "1.5":
+			f.FeePerByte = typed.HalfHourFee
+		case "2":
+			f.FeePerByte = typed.FastestFee
+		}
+		return &f, 0
+	case responses.LitecoinFeeResponse:
+		typed := t.(responses.LitecoinFeeResponse)
+		f.CalcFee = calcLitecoinFee
+		switch speed {
+		case "1":
+			f.FeePerByte = typed.LowFeePerKb
+		case "1.5":
+			f.FeePerByte = typed.MediumFeePerKb
+		case "2":
+			f.FeePerByte = typed.HighFeePerKb
+		}
+		f.FeePerByte = f.FeePerByte / 1024
+		return &f, 0
+	case responses.EthereumFeeResponse:
+		typed := t.(responses.EthereumFeeResponse)
+		switch speed {
+		case "1":
+			gaz = typed.GasPrice * 10 / 8
+		case "1.5":
+			gaz = typed.GasPrice
+		case "2":
+			gaz = typed.GasPrice + (typed.GasPrice * 10 / 5)
+		}
+		return nil, gaz
+	case responses.TokenFeeResponse:
+		typed := t.(responses.TokenFeeResponse)
+		switch speed {
+		case "1":
+			gaz = typed.GasLimit * 10 / 8
```

Comment:
GasLimit -> GasPrice.
GasLimit it's a max amount of gas that may be used during transaction execution